modules/common.nix

```
@@ -61,6 +61,22 @@
           description = "Path of the source file or directory";
         };
 
+        generator = lib.mkOption {
+          type = nullOr (functionTo options.text.type);
+          default = null;
+          description = "Function that when applied to `value` will create the `text` of the file.";
+          example = literalExpression "lib.generators.toGitINI";
+        };
+
+        value = lib.mkOption {
+          type = nullOr (attrsOf anything);
```

```suggestion
          type = nullOr anything;
```
Maybe a bit more fitting if we don't want this API to *only* work with `lib.generators`-like functions? Like maybe one could take in a list? Feedback welcome ;)

---

Lists are bad for overriding and reading, `attrsOf anything` is fine.